mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraMessageDAO.java

```diff
@@ -212,21 +235,29 @@ private ByteBuffer toByteBuffer(InputStream stream) throws IOException {
             .setUUID(MESSAGE_ID, cassandraMessageId.get()));
     }
 
-    private Pair<MessageWithoutAttachment, Stream<MessageAttachmentRepresentation>> message(Row row, ComposedMessageIdWithMetaData messageIdWithMetaData, FetchType fetchType) {
-        ComposedMessageId messageId = messageIdWithMetaData.getComposedMessageId();
+    private Pair<MessageWithoutAttachment, Stream<MessageAttachmentRepresentation>> message(
+        Row row,
+        FetchType fetchType,
+        ComposedMessageIdWithMetaData composedMessageIdWithMetaData
+    ) {
+        Pair<RawMessageWithoutAttachment, Stream<MessageAttachmentRepresentation>> rawPair = rawMessage(row, fetchType);
+
+        return Pair.of(
```

Comment:
I did something like that twice in this PR. IMO it could be great to add a PairUtil in the same package than OptionalConvertor. This pair util will have two statics methods mapLeft and mapRight.